partner_identification/models/res_partner.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2004-2010 Tiny SPRL http://tiny.be
+# Copyright 2010-2012 ChriCar Beteiligungs- und Beratungs- GmbH
+#             http://www.camptocamp.at
+# Copyright 2015 Antiun Ingenieria, SL (Madrid, Spain)
+#        http://www.antiun.com
+#        Antonio Espinosa <antonioea@antiun.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    id_numbers = fields.One2many(
+        comodel_name="res.partner.id_number",
+        inverse_name="partner_id",
+        string="Identification Numbers",
+    )
+
+    @api.depends("id_numbers")
+    def _compute_identification(self, field_name, category_code):
+        """ Compute a field that indicates a certain ID type.
+
+        Use this on a field that represents a certain ID type. It will compute
+        the desired field as that ID(s).
+
+        This ID can be worked with as if it were a Char field, but it will
+        be relating back to a ``res.partner.id_number`` instead.
+
+        Example:
+
+            .. code-block:: python
+
+            social_security = fields.Char(
+                compute=lambda s: s._compute_identification(
+                    'social_security', 'SSN',
+                ),
+                inverse=lambda s: s._inverse_identification(
+                    'social_security', 'SSN',
+                ),
+                search=lambda s, *a: s._search_identification(
+                    'SSN', *a
+                ),
+            )
+
+        Args:
+            field_name (str): Name of field to set.
+            category_code (str): Category code of the Identification type.
+        """
+        for record in self:
+            id_numbers = record.id_numbers.filtered(
+                lambda r: r.category_id.code == category_code
+            )
+            if not id_numbers:
+                continue
```

Comment:
When we use this method as a compute method for a field, then things break if there's no `id_numbers`. We should do `record[field_name] = False` and then `continue`.